### PR TITLE
Allow stream-switching when re-using a client on new host

### PR DESCRIPTION
**Pull request by ca-johnson**
Resolves a bug where re-using the disk from a client that last synced a different stream will result in the flush to match the old clients revisions doing nothing and the stream switch not syncing files accurately.

Fix is to temporarily switch to the same stream as the old workspace, flush, then switch back to the target stream and proceed as normal.

* Add a `file_2.txt` to `//stream-depot/main` + update fixture unit test to document it
* Add test coverage to ensure it is de-synced when stream switching
* Add a new test for client workspace flushing + stream switching simultaneously and make it pass
* Pull the 'flush-to-match' functionality out into its own function

### CODE REVIEW

**Review by improbable-mattchurch — APPROVED** [review]

File: python/test_perforce.py [hunk 1]
```diff
@@ -188,11 +194,12 @@ def test_fixture(capsys, server):
 
 def test_head(server, tmpdir):
     """Test resolve of HEAD changelist"""
-    repo = P4Repo(root=tmpdir)
-    assert repo.head() == "@6", "Unexpected global HEAD revision"
+    # workspace with no changes in view defaults to global view
+    repo = P4Repo(root=tmpdir, view="//depot/empty_dir/... empty_dir/...")
+    assert repo.head() == "@9", "Unexpected global HEAD revision"
```
Comment (ca-johnson) [inline python/test_perforce.py:199]: driveby improvement to test coverage - @6 is not the global head, it should have been @9 with a workspace view that includes no files/changes

File: python/test_perforce.py [hunk 2]
```diff
@@ -188,11 +194,12 @@ def test_fixture(capsys, server):
...
 
-    repo = P4Repo(root=tmpdir, stream='//stream-depot/main')
-    assert repo.head() == "@2", "Unexpected HEAD revision for stream"
+    repo = P4Repo(root=tmpdir, stream='//stream-depot/dev')
+    assert repo.head() == "@8", "Unexpected HEAD revision for stream"
```
Comment (ca-johnson) [inline python/test_perforce.py:202]: Switched to `dev` stream, since `main` stream now has the most recent change so it not effectively testing wether this resolve to something different to global HEAD
